backend/src/bundles/chat/chat.controller.ts

```diff
@@ -0,0 +1,131 @@
+import { type FastifySessionObject } from '@fastify/session';
+
+import {
+    type ApiHandlerOptions,
+    type ApiHandlerResponse,
+    BaseController,
+} from '~/common/controller/controller.js';
+import { ApiPath } from '~/common/enums/enums.js';
+import { HttpCode, HTTPMethod } from '~/common/http/http.js';
+import { type Logger } from '~/common/logger/logger.js';
+
+import { MAX_TOKEN } from './libs/constants/max-token.constant.js';
+import { ChatPath, OpenAIRole } from './libs/enums/enums.js';
+import { type GenerateTextRequestDto } from './libs/types/types.js';
+import { textGenerationValidationSchema } from './libs/validation-schemas/validation-schemas.js';
+import { type OpenAIService } from './open-ai.service.js';
+
+class ChatController extends BaseController {
+    private openAIService: OpenAIService;
+
+    public constructor(logger: Logger, openAIService: OpenAIService) {
+        super(logger, ApiPath.CHAT);
+
+        this.openAIService = openAIService;
+
+        this.addRoute({
+            path: ChatPath.ROOT,
+            method: HTTPMethod.PUT,
```

Comment:
Also why did you decide to use put method for generating chat answer? Isn't post method here more appropriate?